src/components/forms/profile/user-profile-form.tsx

```diff
@@ -15,14 +17,41 @@ import { fetchJson } from "@src/utils/fetch-json";
 
 import "react-datepicker/dist/react-datepicker.css";
 
+type ProfilePayload = Omit<Profile, "id" | "sub" | "updated_at">;
+
+async function updateOrCreateProfile(
+  router: NextRouter,
+  profilePayload: ProfilePayload,
+  isCreation?: boolean,
+): Promise<void> {
+  const url = "/api/profile/user-profile";
+  const method = isCreation ? "POST" : "PATCH";
+  console.log({ method });
+
+  return fetchJson(url, method, profilePayload).then(async (response) => {
+    const parsedResponse = await response.json();
+
+    if (
+      response.status === HttpStatus.CREATED ||
+      response.status === HttpStatus.OK
+    ) {
+      mutate("/api/profile/user-profile", parsedResponse);
+      router && router.push("/account/profile");
+      return;
+    } else {
+      throw new Error("Something went wrong");
+    }
+  });
+}
+
 const UserProfileForm: React.FC<{
   userProfileData?: Profile;
   isCreation?: boolean;
 }> = ({ userProfileData, isCreation }) => {
   const { formatMessage } = useIntl();
   const [formID] = React.useState<string>(uuidv4());
   const [userProfile, setUserProfile] = React.useState<ProfileData>({
-    name: "",
+    name: "Clem",
     family_name: "",
```

Comment:
I guess you can remove this too.

Reply:
😅

Reply:
I took me like 2 mins to understand what was that :laughing:

Reply:
You know... Highly sure and perfectly working manual testing 😁

Reply:
🤣